AdblockPlusSafari/FilterList.m

```diff
@@ -45,16 +45,24 @@ + (NSArray *)boolProperties
 
 - (instancetype)initWithDictionary:(NSDictionary *)dictionary;
 {
+  if (!dictionary) {
+    return nil;
+  }
+
   if (self = [super init]) {
     for (NSString *key in [FilterList allProperties]) {
       [self setValue:[dictionary valueForKey:key] forKey:key];
     }
   }
+
   return self;
 }
 
 - (NSDictionary *)dictionary
 {
+  NSAssert(self.fileName, @"Filename must be set!");
```

Comment:
Assert will do nothing on release run. Is this really just a debug sanity check? Are the asserts always true on release? If not, what will the function produce?